controller/configentry_controller_test.go

```diff
@@ -1104,3 +1129,228 @@ func TestConfigEntryControllers_setsSyncedToTrue(t *testing.T) {
+					Log:    logger,
+					ConfigEntryController: &ConfigEntryController{
+						ConsulClient:   consulClient,
+						DatacenterName: DatacenterName,
+					},
+				}
+			},
+		},
+		{
+			kubeKind:   "ServiceSplitter",
+			consulKind: capi.ServiceSplitter,
+			consulPrereq: []capi.ConfigEntry{
+				&capi.ServiceConfigEntry{
+					Kind:     capi.ServiceDefaults,
+					Name:     "foo",
+					Protocol: "http",
+				},
+			},
+			configEntryResourceWithDeletion: &v1alpha1.ServiceSplitter{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "foo",
+					Namespace:         kubeNS,
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+					Finalizers:        []string{FinalizerName},
+				},
+				Spec: v1alpha1.ServiceSplitterSpec{
+					Splits: []v1alpha1.ServiceSplit{
+						{
+							Weight: 100,
+						},
+					},
+				},
+			},
+			reconciler: func(client client.Client, consulClient *capi.Client, logger logr.Logger) testReconciler {
+				return &ServiceSplitterController{
+					Client: client,
+					Log:    logger,
+					ConfigEntryController: &ConfigEntryController{
+						ConsulClient:   consulClient,
+						DatacenterName: DatacenterName,
+					},
+				}
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.kubeKind, func(t *testing.T) {
+			req := require.New(t)
+
+			s := runtime.NewScheme()
+			s.AddKnownTypes(v1alpha1.GroupVersion, c.configEntryResourceWithDeletion)
+			client := fake.NewFakeClientWithScheme(s, c.configEntryResourceWithDeletion)
+
+			consul, err := testutil.NewTestServerConfigT(t, nil)
+			req.NoError(err)
+			defer consul.Stop()
+			consulClient, err := capi.NewClient(&capi.Config{
+				Address: consul.HTTPAddr,
+			})
+			req.NoError(err)
+
+			// Create any prereqs.
+			for _, configEntry := range c.consulPrereq {
+				written, _, err := consulClient.ConfigEntries().Set(configEntry, nil)
+				req.NoError(err)
+				req.True(written)
+			}
+
+			// We haven't run reconcile yet so we must create the config entry
+			// in Consul ourselves.
+			{
+				// Create the resource with differnt datacenter on metadata
+				written, _, err := consulClient.ConfigEntries().Set(c.configEntryResourceWithDeletion.ToConsul("different-datacenter"), nil)
+				req.NoError(err)
+				req.True(written)
+			}
+
+			// Now run reconcile. It's marked for deletion so this should delete the kubernetes resource
+			// but not the consul config entry.
+			{
+				namespacedName := types.NamespacedName{
+					Namespace: kubeNS,
+					Name:      c.configEntryResourceWithDeletion.Name(),
+				}
+				r := c.reconciler(client, consulClient, logrtest.TestLogger{T: t})
+				resp, err := r.Reconcile(ctrl.Request{
+					NamespacedName: namespacedName,
+				})
+				req.NoError(err)
+				req.False(resp.Requeue)
+
+				entry, _, err := consulClient.ConfigEntries().Get(c.consulKind, c.configEntryResourceWithDeletion.Name(), nil)
+				req.NoError(err)
+				req.Equal(entry.GetMeta()[common.DatacenterKey], "different-datacenter")
+			}
+		})
+	}
+}
```

Comment:
Also need a test that if it exists in another DC that it doesn't modify the resource in consul and it sets its status accordingly.

Reply:
writing that test as we speak :)